packages/edit-widgets/src/store/actions.js

```diff
@@ -120,13 +120,20 @@ export function* saveWidgetArea( widgetAreaId ) {
 		return true;
 	} );
 
-	// Get all widgets that have been deleted
-	const deletedWidgets = areaWidgets.filter(
-		( { id } ) =>
-			! widgetsBlocks.some(
-				( widgetBlock ) => getWidgetIdFromBlock( widgetBlock ) === id
-			)
-	);
+	// Determine which widgets have been deleted. We can tell if a widget is
+	// deleted and not just moved to a different area by looking to see if
+	// getWidgetAreaForWidgetId() finds something.
+	const deletedWidgets = [];
+	for ( const widget of areaWidgets ) {
+		const widgetsNewArea = yield select(
+			editWidgetsStoreName,
+			'getWidgetAreaForWidgetId',
```

Comment:
This is surprising to me that `getWidgetAreaForWidgetId` actually uses `getEditedEntityRecord` under the hood so it's returning the latest edited state rather than the stored one. Is it intentional? Should we make it more explicit?

Reply:
Hm yes probably it shouldn't. What do you mean by make it more explicit? Happy to accept a ```suggestion 😛

Reply:
Oh I mean to rename `getWidgetAreaForWidgetId` to something like `getEditedWidgetAreaForWidgetId`.

Reply:
Hm, we can't simply rename because it's a public function. We'd have to deprecate `getWidgetAreaForWidgetId` and add `getEditedWidgetAreaForWidgetId`. I'm not sure if it's worth it? What do you think?

I kind of want to change how all of this works after 5.8... I don't think we should have `select( 'core/edit-widgets' ).getWidgetAreaForWidgetId()` at all because we already have e.g. `select( 'core' ).getWidgetArea()` which can do the job if we add `blocks` to the widget area entity type.

Reply:
Sure! Whatever you think is the best 👍